src/utils/file-path.ts

```diff
@@ -0,0 +1,56 @@
+import * as R from 'ramda'
+import type {RuleParamsBase} from './types'
+
+/**
+ * For all created/modified files finds those starting with includePaths (excluding those starting with excludePaths).
+ *
+ * @param danger Danger instance
+ * @param excludePaths paths to exclude
+ * @param includePaths paths to include
+ */
+export const filterPaths = (params: Omit<RuleParamsBase, 'fail'>): Array<string> => {
+  const {danger, excludePaths = [], includePaths} = params
+
+  return R.compose(
+    R.map((file: string) => {
+      const {created, edited} = danger.git.fileMatch(file).getKeyedPaths()
+      return created[0] || edited[0]
+    }),
+    R.reject(
+      R.anyPass([
+        ...R.map((includePath) => R.compose(R.not, R.startsWith(includePath)), includePaths),
+        ...R.map<string, (str: string) => boolean>(R.startsWith, excludePaths),
+      ]),
+    ),
+  )(R.concat(danger.git.modified_files, danger.git.created_files))
+}
+
+/**
+ * For the provided file paths traverses up through all containing folders
```

Comment:
@quesabe can you please explain what the function does in a more verbose way? Does it exclude some paths from the list of paths passed in and then returns a set of unique paths?

Reply:
Ah, managed to figure out by reading the tests above. Let's maybe change the description to this one?
```
Returns a set of unique paths which are parents to all the paths from `paths` input param.
```

What do you think?